orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/ClusterMatcher.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.clouddriver.utils;
+
+import com.netflix.frigga.Names;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class ClusterMatcher {
+
+  public static ClusterMatchRule getMatchingRule(String account, String location, String clusterName, List<ClusterMatchRule> rules) {
+    if (rules == null || rules.isEmpty()) {
+      return null;
```

Comment:
Could use `Optional` and replace the null check with `.isPresent()`

Reply:
sure